api/client/cloud/cloud_test.go

```diff
@@ -209,196 +180,197 @@ func (s *cloudSuite) TestClouds(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 	c.Assert(result, gc.IsNil)
-	c.Assert(s.called, jc.IsTrue)
 }
 
 func (s *cloudSuite) TestUpdateCredentialError(c *gc.C) {
-	apiCaller := basetesting.BestVersionCaller{
-		APICallerFunc: basetesting.APICallerFunc(
-			func(objType string,
-				version int,
-				id, request string,
-				a, result interface{},
-			) error {
-				c.Check(request, gc.Equals, "UpdateCredentialsCheckModels")
-				*result.(*params.UpdateCredentialResults) = params.UpdateCredentialResults{
-					Results: []params.UpdateCredentialResult{
-						{CredentialTag: "cloudcred-foo_bob_bar",
-							Error: apiservererrors.ServerError(errors.New("validation failure")),
-						},
-					},
-				}
-				s.called = true
-				return nil
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	args := params.UpdateCredentialArgs{
+		Credentials: []params.TaggedCredential{{
+			Tag: "cloudcred-foo_bob_bar",
+			Credential: params.CloudCredential{
+				AuthType: "userpass",
+				Attributes: map[string]string{
+					"username": "admin",
+					"password": "adm1n",
+				},
+			},
+		}}}
+	res := new(params.UpdateCredentialResults)
+	results := params.UpdateCredentialResults{
+		Results: []params.UpdateCredentialResult{
+			{
+				CredentialTag: "cloudcred-foo_bob_bar",
+				Error:         apiservererrors.ServerError(errors.New("validation failure")),
 			},
-		),
-		BestVersion: 3,
+		},
 	}
-	client := cloudapi.NewClient(apiCaller)
+	mockFacadeCaller := basemocks.NewMockFacadeCaller(ctrl)
+	mockFacadeCaller.EXPECT().FacadeCall("UpdateCredentialsCheckModels", args, res).SetArg(2, results).Return(nil)
+	client := cloudapi.NewClientFromCaller(mockFacadeCaller)
+
 	errs, err := client.UpdateCredentialsCheckModels(testCredentialTag, testCredential)
 	c.Assert(err, gc.ErrorMatches, "validation failure")
 	c.Assert(errs, gc.IsNil)
-	c.Assert(s.called, jc.IsTrue)
 }
 
 func (s *cloudSuite) TestUpdateCredentialCallError(c *gc.C) {
-	apiCaller := basetesting.BestVersionCaller{
-		APICallerFunc: basetesting.APICallerFunc(
-			func(objType string,
-				version int,
-				id, request string,
-				a, result interface{},
-			) error {
-				c.Check(request, gc.Equals, "UpdateCredentialsCheckModels")
-				s.called = true
-				return errors.New("scary but true")
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	args := params.UpdateCredentialArgs{
+		Credentials: []params.TaggedCredential{{
+			Tag: "cloudcred-foo_bob_bar",
+			Credential: params.CloudCredential{
+				AuthType: "userpass",
+				Attributes: map[string]string{
+					"username": "admin",
+					"password": "adm1n",
+				},
+			},
+		}}}
+	res := new(params.UpdateCredentialResults)
+	results := params.UpdateCredentialResults{
+		Results: []params.UpdateCredentialResult{
+			{
+				CredentialTag: "cloudcred-foo_bob_bar",
+				Error:         apiservererrors.ServerError(errors.New("validation failure")),
 			},
-		),
-		BestVersion: 3,
+		},
```

Comment:
removing the `BestVersion: 3`, the scope of the test has changed. It appears that version 3 of the facade was removed in juju 3.0. Is `TestUpdateCredentialCallError` really different from `TestUpdateCredentialError`?

Reply:
Same for any test where `BestVersion: 3` has been removed.